transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -32,9 +33,14 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
   // format is: metadata/{entity_type}/{entity_id}_{version}.json,
   // but filename returns just {entity_id}_{version}.json, so that is what we deal with.
   val metadataPattern: Regex = "([^_]+)_(.+).json".r
-  // format is: {dir_path}/{file_id}_{file_version}_{file_name},
+  // format is: data/{dir_path}/{file_id}_{file_version}_{file_name},
   // but the dir_path seems to be optional
   val fileDataPattern: Regex = "(.*\\/)?([^_^\\/]+)_([^_]+)_(.+)".r
+  // format is: links/{links_id}_{version}_{project_id}.json
+  // but the `project_id` identifies the project the subgraph is part of
+  // A subgraph is part of exactly one project. The importer must record an error if it
+  // detects more than one object with the same `links/{links_id}_{version}_` prefix.
+  val linksDataPattern: Regex = "([^_]+)_(.+)_([^_]+).json".r
```

Comment:
The [design doc](https://docs.google.com/document/d/1AJfpaGMauFlusgAOKQHNeCta-PrHz5OObO-FcHyn_zw/edit?usp=sharing) is suggesting that the format for the links filenames should just be `{links_id}_{version}.json`  and the project_id part should come from the JSON content (under `project_id`).

Maybe the doc is out of date though, do you know if something changed?

Reply:
Oh, nevermind, it did change. This looks good :)

Reply:
Ah, I think I missed that first version. That's a good point though, I'll make sure to keep double checking the design document for more changes.